Anagram.java

```diff
@@ -37,13 +87,116 @@ public static boolean isAnagram(String str1, String str2) {
 	// as is. For example, the string "What? No way!" becomes "whatnoway"
 	public static String preProcess(String str) {
 		// Replace the following statement with your code
-		return "";
+
+		String newString = "";
+		char newLettar;
+		for(int i = 0; i<str.length(); i++){
+			if ((((int)str.charAt(i)) < 97 || ((int)str.charAt(i))>122)) {
+				if (((int)str.charAt(i)) > 64 && (int)str.charAt(i) < 96) {
+					newLettar = (char)(((int)str.charAt(i))+32);
+					newString = newString + newLettar;
+				}
+				if ((int)str.charAt(i) == 32){
+					newString = newString + str.charAt(i);
+				}
+			}
+
+			else{
+				newString = newString + str.charAt(i);
+			}
+		}
+			return newString;
 	} 
 	   
 	// Returns a random anagram of the given string. The random anagram consists of the same
 	// characters as the given string, re-arranged in a random order. 
 	public static String randomAnagram(String str) {
 		// Replace the following statement with your code
-		return "";
+		String newStr = "";
+		double randnum = Math.random()*str.length();
+		for(int i = 0; i<str.length(); i++)
+		{
+			if (str.charAt((int)randnum) != ' ') {
+				newStr = newStr + str.charAt((int) randnum);
+				str = str.replace(str.charAt((int)randnum), ' ');
```

Comment:
str.replace() wasn't taught in lectures or recitations. HW guidelines violation -2